examples/java/src/main/java/org/apache/beam/examples/complete/datatokenization/transforms/io/TokenizationFileSystemIO.java

```diff
@@ -141,108 +129,133 @@
 
   private final DataTokenizationOptions options;
 
-  public FileSystemIO(DataTokenizationOptions options) {
+  public TokenizationFileSystemIO(DataTokenizationOptions options) {
     this.options = options;
   }
 
-  public PCollection<String> read(Pipeline pipeline, String schema) {
-    if (options.getInputFileFormat() == FORMAT.JSON) {
-      return pipeline.apply("ReadJsonFromFiles", TextIO.read().from(options.getInputFilePattern()));
-    } else if (options.getInputFileFormat() == FORMAT.CSV) {
-      PCollectionTuple jsons =
-          pipeline
-              /*
-               * Step 1: Read CSV file(s) from File System using {@link CsvConverters.ReadCsv}.
-               */
-              .apply(
-                  "ReadCsvFromFiles",
-                  CsvConverters.ReadCsv.newBuilder()
-                      .setCsvFormat(options.getCsvFormat())
-                      .setDelimiter(options.getCsvDelimiter())
-                      .setHasHeaders(options.getCsvContainsHeaders())
-                      .setInputFileSpec(options.getInputFilePattern())
-                      .setHeaderTag(CSV_HEADERS)
-                      .setLineTag(CSV_LINES)
-                      .build())
-              /*
-               * Step 2: Convert lines to Json.
-               */
-              .apply(
-                  "LineToJson",
-                  CsvConverters.LineToFailsafeJson.newBuilder()
-                      .setDelimiter(options.getCsvDelimiter())
-                      .setJsonSchema(schema)
-                      .setHeaderTag(CSV_HEADERS)
-                      .setLineTag(CSV_LINES)
-                      .setUdfOutputTag(PROCESSING_OUT)
-                      .setUdfDeadletterTag(PROCESSING_DEADLETTER_OUT)
-                      .build());
-
-      if (options.getNonTokenizedDeadLetterPath() != null) {
-        /*
-         * Step 3: Write jsons to dead-letter that weren't successfully processed.
-         */
-        jsons
-            .get(PROCESSING_DEADLETTER_OUT)
-            .apply(
-                "WriteCsvConversionErrorsToFS",
-                ErrorConverters.WriteStringMessageErrorsAsCsv.newBuilder()
-                    .setCsvDelimiter(options.getCsvDelimiter())
-                    .setErrorWritePath(options.getNonTokenizedDeadLetterPath())
-                    .build());
-      }
+  public PCollection<Row> read(Pipeline pipeline, SchemasUtils schema) {
+    switch (options.getInputFileFormat()) {
```

Comment:
where is Avro? we did not implement Avro in the beam example? @daria-malkova @KhaninArtur

Reply:
Nope, we didn't, I think it can be done after merging of this PR